kubernetes-extension-test/src/test/java/org/ballerinax/kubernetes/test/IstioGatewayTest.java

```diff
@@ -0,0 +1,487 @@
+        Assert.assertEquals(port.get("number"), 80, "Invalid port number.");
+        Assert.assertEquals(port.get("name"), "http", "Invalid port name.");
+        Assert.assertEquals(port.get("protocol"), "HTTP", "Invalid port protocol.");
+
+        List<String> hosts = (List<String>) server.get("hosts");
+        Assert.assertTrue(hosts.contains("uk.bookinfo.com"), "uk.bookinfo.com host not included");
+        Assert.assertTrue(hosts.contains("eu.bookinfo.com"), "eu.bookinfo.com host not included");
+
+        Map<String, Object> tls = (Map<String, Object>) server.get("tls");
+        Assert.assertEquals(tls.get("httpsRedirect"), true, "Invalid tls httpsRedirect value");
+
+        KubernetesUtils.deleteDirectory(targetPath);
+        KubernetesTestUtils.deleteDockerImage(dockerImage);
+    }
+    
+    /**
+     * Build bal file with invalid port istio gateway annotations.
+     *
+     * @throws IOException          Error when loading the generated yaml.
+     * @throws InterruptedException Error when compiling the ballerina file.
+     */
+    @Test
+    public void invalidPortTest() throws IOException, InterruptedException {
+        Assert.assertEquals(KubernetesTestUtils.compileBallerinaFile(balDirectory, "invalid_port.bal"), 1);
+    }
+    
+    /**
+     * Build bal file with istio gateway annotation having multiple servers.
+     *
+     * @throws IOException               Error when loading the generated yaml.
+     * @throws InterruptedException      Error when compiling the ballerina file.
+     * @throws KubernetesPluginException Error when deleting the generated artifacts folder.
+     */
+    @Test
+    public void multipleServersTest() throws IOException, InterruptedException, KubernetesPluginException {
+        Assert.assertEquals(KubernetesTestUtils.compileBallerinaFile(balDirectory, "multiple_servers.bal"), 0);
+
+        // Check if docker image exists and correct
+        validateDockerfile();
+        validateDockerImage();
+
+        // Validate gateway yaml
+        File gatewayFile = Paths.get(targetPath).resolve("multiple_servers_istio_gateway.yaml").toFile();
+        Assert.assertTrue(gatewayFile.exists());
+        Yaml yamlProcessor = new Yaml();
+        Map<String, Object> gateway = (Map<String, Object>) yamlProcessor.load(FileUtils.readFileAsString(gatewayFile));
+        Assert.assertEquals(gateway.get("apiVersion"), "networking.istio.io/v1alpha3", "Invalid apiVersion");
+        Assert.assertEquals(gateway.get("kind"), "Gateway", "Invalid kind.");
+
+        Map<String, Object> metadata = (Map<String, Object>) gateway.get("metadata");
+        Assert.assertEquals(metadata.get("name"), "my-gateway", "Invalid gateway name");
+
+        Map<String, Object> spec = (Map<String, Object>) gateway.get("spec");
+        Map<String, Object> selector = (Map<String, Object>) spec.get("selector");
+        Assert.assertEquals(selector.get("app"), "my-gateway-controller", "Invalid selector.");
```

Comment:
Shall we use `KubernetesConstants.KUBERNETES_SELECTOR_KEY` here?

Reply:
Fixed with https://github.com/ballerinax/kubernetes/pull/220/commits/5f8887c1388816ebe18d4066c627a7d43a88bea9